src/auth/User.h

```diff
@@ -71,6 +71,16 @@ class User : public RefCountable
      */
     virtual int32_t ttl() const = 0;
 
+    /// simple accessor to detect expired credentials
```

Comment:
Removing the comment violates the requirement to provide a description of what each public method does.

Suggestions for better wording are welcome.

Reply:
After looking at a few similar trivial methods accepted in other PR I'm thinking something like
  "whether the credentials represented by this Auth::User object have passed their latest known TTL"

Reply:
> Removing the comment violates the requirement to provide a description of what each public method does.

In my [documented](https://github.com/measurement-factory/squid-notes/tree/start#doc2-there-are-no-absolute-principles) opinion, there are virtually no absolute requirements. The "every new name should be documented" (or equivalent) rule does apply to this method, of course, but leaving this trivial test method without explicit documentation would be reasonable IMO if no suitable documentation is found (i.e. exceptional treatment can be justified in this specific case). Providing proper documentation would be even better, of course, but the current "foo is foo" documentation is (slightly) worse than the absence of any explicit documentation.

> "whether the credentials represented by this Auth::User object have passed their latest known TTL"

AFAICT, the only useful bit of information here is the word "credentials". The rest is essentially a definition of what the word "expires" means in a context with a known TTL. By saying "credentials", are we implying that the user record itself (i.e. the Auth::User object) does not really expire and can be meaningfully used by Squid beyond the credentials expiration time? If yes, let's be very explicit about that important fact -- it is valuable. Otherwise, let's not imply that credentials have a separate "usable lifetime" from their User object.

Reply:
> > "whether the credentials represented by this Auth::User object have passed their latest known TTL"
> 
> AFAICT, the only useful bit of information here is the word "credentials".

That is somewhat insulting. But i shall overlook other than to point out.

> The rest is essentially a definition of what the word "expires" means in a context with a known TTL.

Exactly. Documentating what "expired()" means in its use-case context(s).

> By saying "credentials", are we implying that the user record itself (i.e. the Auth::User object) does not really expire and can be meaningfully used by Squid beyond the credentials expiration time?

Yes. The object itself only "expires" when free()'d. The credentials/value can be expired/stale without free().

> If yes, let's be very explicit about that important fact -- it is valuable.

Suggestions welcome. I thought that original wording sufficiently clear "to detect expired credentials" (note word is *credentials*, not "user")